### PR TITLE
Move data entry functionality to new module

### LINKED ISSUE #380: Backend: move data entry to new module
The backend currently has a 'polling station' module that contains code for both polling stations and data entry. We should move the data entry code to a dedicated module, so that we have a clear separation once we start adding polling station administration code (adding, editing and deleting polling stations).

### PR DESCRIPTION
**Pull request by praseodym**
Resolves #380

### CODE REVIEW

File: backend/src/polling_station/repository.rs [hunk 1]
```diff
@@ -1,16 +1,12 @@
...
 use crate::AppState;
 
-use super::{
-    PollingStation, PollingStationResults, PollingStationResultsEntry, PollingStationStatusEntry,
-};
-
 pub struct PollingStations(SqlitePool);
 
 impl PollingStations {
-    #[cfg(test)]
```
Comment (praseodym) [inline backend/src/polling_station/repository.rs:13]: Needed to remove it to get this to work:
https://github.com/kiesraad/abacus/blob/62473a4e7add806a27ee03aea205007aa4005fe8/backend/src/polling_station/repository.rs#L235

I'm going to solve that in a better way by passing the `PollingStation` repository directly instead.

Comment (praseodym) [reply backend/src/polling_station/repository.rs]: Changed in 635e819432414e17ecc1138c9bd7f721fcbc8500.

File: backend/src/lib.rs [hunk 2]
```diff
@@ -59,7 +60,7 @@ pub fn router(pool: SqlitePool) -> Result<Router, Box<dyn Error>> {
 
     let app = Router::new()
         .nest("/api/elections", election_routes)
-        .nest("/api/polling_stations", polling_station_routes);
+        .nest("/api/polling_stations", data_entry_routes);
```
Comment (oliver3) [inline backend/src/lib.rs:63]: This now looks a bit inconsistent. Maybe we can change the url to include the `/:polling_station_id/data_entries/` here, or keep the polling_station_routes and nest data_entry_routes in there again?

Comment (praseodym) [reply backend/src/lib.rs]: Good idea, changed in 19ae4988967a1f8cb48aa3453b42e6679e129e6e.

**Review by oliver3 — APPROVED** [review]

**Review by Lionqueen94 — APPROVED** [review]

**Pull request merged**